### ISSUE TITLE
Forge hammer bug

### ISSUE THREAD

**Issue by Aquiles089** [Syrikal/Tetranomicon#13]
The 1.19 update has a very big bug that requires urgent repair! When you have Tetranomicon installed you cannot use the Forge hammer, you can even use cells and upgrades, but it doesn't work as a hammer.

**Comment by mickelus** [comment]
This is due to https://github.com/Syrikal/Tetranomicon/blob/main/src/main/resources/data/tetra/tiers/vanilla.json overriding the default tiers tetra cares for without including "tetra:maxed_forge_hammer". The tiers data tells tetra which tiers it should care for (and does not affect how they are ordered), add a new file in the tiers folder if you want to add additional tiers and it will pick those up. Tetra references the forge `TierSortingRegistry` and does not register tiers on it's own, that registry also handles the ordering of the tiers. Tetranomicon does not seem to register its tiers, adding the following to your mod constructor would probably solve that:

```java
TagKey<Block> sixTag = BlockTags.create(new ResourceLocation("tetranomicon:needs_tier_six_tool"));
Tier sixTier = TierSortingRegistry.registerTier(new ForgeTier(TetraRegistries.forgeHammerTier.getLevel() + 1, 0, 0, 0, 0,
        sixTag, () -> Ingredient.EMPTY), new ResourceLocation("tetranomicon:tier_six"), List.of(TetraRegistries.forgeHammerTier), List.of());
TagKey<Block> sevenTag = BlockTags.create(new ResourceLocation("tetranomicon:needs_tier_seven_tool"));
Tier sevenTier = TierSortingRegistry.registerTier(new ForgeTier(sixTier.getLevel() + 1, 0, 0, 0, 0,
        sevenTag, () -> Ingredient.EMPTY), new ResourceLocation("tetranomicon:tier_seven"), List.of(sixTier), List.of());
```

I can make a PR with a fix, let me know if that's of interest!

**Comment by Aquiles089** [comment]
> I can make a PR with a fix, let me know if that's of interest!

That would be a dream, i would love if you made the pr!
That's why mickelus is the goat.

**Comment by mickelus** [comment]
That was for syric :P

**Comment by Syrikal** [comment]
Ah gotcha! I'm not used to the new tiers system, so I thought it was a Tetra-specific thing. I'll get this fixed up.

**Comment by Syrikal** [comment]
> That was for syric :P

Incidentally, is this fix also needed in 1.18? It also uses the new tier system, but there's no Tetra forgeHammerTier in that version.

**Comment by Aquiles089** [comment]
Only testing may prove that

**Comment by Aquiles089** [comment]
Only testing may prove that

**Comment by mickelus** [comment]
It should be fine in 1.18!

**Comment by Syrikal** [comment]
Fixed in 1.4.2.
